test/e2e/collect-ci-artifacts.sh

```diff
@@ -7,7 +7,10 @@ set -o errexit
 : "${KUBECONFIG:?}"
 : "${TEST_NAMESPACE:?}"
 : "${TEST_ARTIFACTS_DIR:?}"
+: "${KUBECTL:=kubectl}"
```

Comment:
Note: this KUBECTL variable defaults to kubectl when it hasn't been defined/overridden locally. This change now allows users to override this variable in their local/CI environments, e.g. `make e2e KUBECTL=oc`.